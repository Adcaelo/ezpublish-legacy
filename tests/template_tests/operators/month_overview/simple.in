{use $month_overview_data}
{var $month = month_overview($month_overview_data, 'published', 1193446660)}
{raw tc_access('simple.tpl:0',  $month, 'year')}
{foreach tc_make_array('simple.tpl:2', tc_access('simple.tpl:1',  $month, 'weekdays')) as $weekday} 
{tc_access('simple.tpl:3',  $weekday, 'day')}
{/foreach}
<br/>
{raw tc_access('simple.tpl:4',  $month, 'month')}




