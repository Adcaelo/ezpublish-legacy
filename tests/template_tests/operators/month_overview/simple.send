<?php

class MonthOverviewData
{
    private $timestamp;

    public function __construct($timestamp)
    {
        $this->timestamp = $timestamp;
    }

    public function hasAttribute($name)
    {
        return $name == "timestamp";
    }

    public function attribute($name)
    {
        if( $this->hasAttribute($name) )
        {
            return $this->timestamp;
        }
        return null;
    }
}

// Create some input data.
$m = array(
    new MonthOverviewData(1194446624),
    new MonthOverviewData(1194346624),
    new MonthOverviewData(1194246624),
    new MonthOverviewData(1194146624),
    new MonthOverviewData(1194046624),
    new MonthOverviewData(1193946624),
    new MonthOverviewData(1193846624),
    new MonthOverviewData(1193746624),
    new MonthOverviewData(1193646624),
    new MonthOverviewData(1193546624),
    new MonthOverviewData(1193446624),
    new MonthOverviewData(1193346624),
);


return array("month_overview_data" => $m);
?>
