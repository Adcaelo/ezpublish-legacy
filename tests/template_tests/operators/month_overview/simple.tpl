{let month=$month_overview_data|month_overview('published', 1193446660)}
{$month.year}

{section var=weekday loop=$month.weekdays}
{$weekday.item.day}

{/section}
<br/>
{$month.month}

{/let}
