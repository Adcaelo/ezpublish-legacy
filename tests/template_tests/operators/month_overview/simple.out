2007
Sun
Mon
Tue
Wed
Thu
Fri
Sat
<br/>
October
